pkg/hoist/hoist_launchable.go

```diff
@@ -359,3 +342,21 @@ func (hl *Launchable) InstallDir() string {
 	launchableName := hl.Version()
 	return filepath.Join(hl.RootDir, "installs", launchableName)
 }
+
+func (hl *Launchable) p2ExecArgs() []string {
```

Comment:
Considering how this method is used, I think it makes sense to have it build the entire command.  Build the slice with `hl.P2exec` prepended, and add a `command string` parameter that gets appended. Then there will be no more need for the funky `append()` calls at the method call sites.